### ISSUE TITLE
How do I host the PokemonGo-Map server outside just localhost? I.e. how do I reach it from other devices or from the internet?

### ISSUE THREAD

**Issue by hkuiyo** [AHAAAAAAA/PokemonGo-Map#2831]
I'm trying to forward port 5000 (remote) to my laptop running PokemonGo-Map on port 5000.
But I'm unable to reach it. Also unable to reach it using http://192.168.0.100:5000/ only http://localhost:5000/ works.

**Comment by MRokas** [comment]
Get a VPN, or deploy to Heroku.

**Comment by RowanF1** [comment]
You need to put your WAN ip adress in the command line.

**Comment by coderkd10** [comment]
have a look at this guide https://github.com/Langoor2/PokemonGo-Map-FAQ/blob/master/FAQ/Portforwarding.md

**Comment by tkSimon** [comment]
Or check out the DigitalOcean or Heroku hosting wiki at https://www.jonaharagon.com/PoGoMapWiki Installation/Advanced.

**Comment by chromer90** [comment]
Change the host to 0.0.0.0 in the config or use -H 0.0.0.0 in your command. Works for me.

**Comment by ro4sho** [comment]
Indeed, use -H 0.0.0.0

**Comment by peshetoman** [comment]
easy one click setup is with ngrok. but the link changes each time you rin ngrok and also it is limited 40 requests per minute if you exceed you need to restart ngrok (via teamviewer)

**Comment by adosikas** [comment]
Actually if you want IPv6 connections as well you have to use "-H ::" (see #2841)

**Comment by pankk** [comment]
To summarize: use use -H 0.0.0.0 _and_ make sure your port is forwarded in your router _and_ be aware that if you are currently connected to the same network that the server is on (WiFi on your phone) it won't work (so turn off WiFi if you are testing).

**Comment by ZankerZCT** [comment]
I still have the issue with host externally. I followed the instructions 100% and done every troubleshooting I can think of. To be honest, when I type in -H 0.0.0.0 in command line and run it. It gives no error or anything. I'm running it in debug mode to see if I can catch anything unusual. Nothing so far. Checked over 10+ port check tools available over internet and all of them confirmed my forwarded ports. Whenever I type in my WAN IP with port. It just didn't connect. All I got is " 'This' taking too long to response."

My apologies in advance if it is fixed or a solution is offered.

**Comment by chromer90** [comment]
@zankerzct what happens if you do your local IP then the port ie: 192.168.1.120:5012

**Comment by ZankerZCT** [comment]
Nothing happens. Map works. Pokemon scan works. Everything works. Just externally connecting to this PC via WAN IP just don't work. I'm forced to use ngrok until this gets resolved and I'd rather not using ngrok.

**Comment by chromer90** [comment]
So what are you going to to access the map? Are you still using localhost:5000?

Generally the command should look like this:

`python runserver.py -a google -u USER -p PASS -l "Some Location" -st 10 -P 5009  -k GoogleMapKey -H 0.0.0.0`

the -P changes from the default 5000 to whatever port you want. This is not necessary.

If you are doing it exactly like that AND have the port forwarded in your router to the local IP of your computer running the server it should just work.

**Comment by ZankerZCT** [comment]
I'm sorry, but it still don't work.

I also made a script that automatically runs the search and ngrok for me. My python command looks like this. (Separate script for my regular scan.)

`cd C:\PokemonGo-Map-master\
python runserver.py -a google -u censored -p censored -st 10 -t 3 -sd 2 -ng -nk -k "Google Maps Key" -l "censored" -H 0.0.0.0 -P 5002
pause`

Pause is there to stop command window from closing down so I can read the errors and etc.

**Comment by chromer90** [comment]
well unfortunately you are doing nothing wrong. Only other thing I could recommend would be to try doing it from the config. Should be the very last options at the bottom.
```
# Webserver settings
host:                  0.0.0.0
#port:                  # port to listen on (default 5000)
```

**Comment by ZankerZCT** [comment]
I already tried that as well. It doesn't work as well. :P

**Comment by chromer90** [comment]
Well beats the heck out of me then lol. Last resort would be delete everything and backup your scripts and then redownload and start over and see if it works again. Worked for me every time and I am running like 20 different locations all accessible remotely.

**Comment by amistad18** [comment]
I'm using Ngrok:
https://ngrok.com/

Just unzip it and run:
`ngrok http 5000`

It's that easy ;)
The only issue for me, is the subdomain of ngrok.io changes every time when you run ngrok command

**Comment by ZankerZCT** [comment]
I already mentioned that I prefer not using ngrok for that particular reason, amistad. Thanks for replying.

@chromer90, yeah. Beat me too... I'm looking into my windows firewall and it still allows it because I'm secured. (My phone is, actually.) and I did redownload and everything. It's my computer that just wouldn't bend to my will. >_>

**Comment by pankk** [comment]
Not advice you will hear me give, but try to disable your firewall and your virus scanner, why do you even have one? Don't do this though. winky-face You could of course also try it on your Raspberry or equivalent, or install a live system on an USB stick to rule out your OS or associated software is the culprit.

**Comment by ZankerZCT** [comment]
Well, here is weird part.

I'm able to connect directly to my WAN IP without ports and it took me to router settings on my phone. It's scary.

I'm not sure if the real issue is the port itself or the host.

**Comment by pankk** [comment]
@ZankerZCT 

> and be aware that if you are currently connected to the same network that the server is on (WiFi on your phone) it won't work (so turn off WiFi if you are testing).

You made me quote myself! What is going on?!
Don't worry about it ;) Nice that it's working at last. Please close the issue.

**Comment by Tagor322** [comment]
@ZankerZCT Hi, can you post a list of all steps you have done to connect via the WAN IP?
Would be really helpful ;)  Thanks!

**Comment by ZankerZCT** [comment]
I think my only issue was so that my WAN IP is not accessible to anyone who is connected to my network, but accessible to anyone else who isn't. That's why I posted and looking for a solution when it's soooo obvious in my eyes.

The steps to do this is already provided in wiki. I just can't provide my own steps because I dabbled too much around it and the wiki just lacked this information, that's all.

**Comment by Tagor322** [comment]
Okay :D Did you insert your WAN IP into the Config or the Commandline/Batch?

**Comment by ZankerZCT** [comment]
No sir. Just -H 0.0.0.0 to command line. Adding 0.0.0.0 IP to host line in config works too.

**Comment by Tagor322** [comment]
Thanks! And then you entert your WAN IP into the iPokeGp App and it works?

**Comment by ZankerZCT** [comment]
No. You don't need any apps for it to work. Need any browser with google maps enabled.

WANIP:Port should do in your browser.

**Comment by Tagor322** [comment]
Okay. I try this later with the iOS App and hope it works :D Thanks

**Comment by ZankerZCT** [comment]
If you're using the scripts on ipad/iphone/whatsoever and using that to scan the area, you don't need to use -H 0.0.0.0 for that because you can easily access them via 127.0.0.1 or localhost

**Comment by Tagor322** [comment]
It works fine in my local network, there are no problems. I want to use the map on my Phone when im not in my local network (in the Town). The iPokeGo App wants a server address and the server must be accessible over 3G or LTE.

**Comment by ZankerZCT** [comment]
Lemme clarify on my part. Did you mean to run the scripts ON your phone? If so, it's beyond me because I haven't root my android phone for that use yet. Sorry, friend.

**Comment by Tagor322** [comment]
There is an Github Projekt linkted under iOS Port. https://github.com/istornz/iPokeGo and there is an Android port too https://github.com/omkarmoghe/Pokemap. I used the iOS version and this app wants an Server for running ;)
